control/control.go

```diff
@@ -225,6 +239,83 @@ func New(cfg *Config) *pluginControl {
 	return c
 }
 
+func (p *pluginControl) HandleGomitEvent(e gomit.Event) {
+	switch v := e.Body.(type) {
+	case *control_event.LoadPluginEvent:
+		// Get all known tasks data
+		for taskID, taskData := range p.taskIDData {
+			// Expand every task metric namespaces (with "*") to corresponding metric calalog namespaces
+			for _, tm := range taskData.metrics {
+
+				matchedNss, err := p.metricCatalog.MatchNamespaces(tm.Namespace())
+
+				if err != nil {
+					log.WithFields(log.Fields{
+						"_block":         "control",
+						"event":          v.Namespace(),
+						"plugin-name":    v.Name,
+						"plugin-version": v.Version,
+						"plugin-type":    core.PluginType(v.Type).String(),
+					}).Error("error matching task namespace with metric catalog")
+				} else {
+					if len(matchedNss) > 0 {
+						depMts := []core.Metric{}
+						for _, ns := range matchedNss {
+
+							// Get expanded namespace data from metric catalog
+							if m, err := p.metricCatalog.Get(ns, tm.Version()); err == nil {
+								// Check if expanded namespace belongs to currently loaded plugin
+								if m.Plugin.TypeName() == core.PluginType(v.Type).String() && m.Plugin.Name() == v.Name {
+									depMts = append(depMts, &metric{
+										namespace: ns,
+										version:   tm.Version(),
+										config:    taskData.configTree.Get(ns.Strings()),
+									})
+								}
+							}
+						}
+
+						// Validate and subscribe deps for loaded plugin
+						errs := p.ValidateDeps(depMts, taskData.plugins)
+						if len(errs) > 0 {
+							log.WithFields(log.Fields{
+								"_block":         "control",
+								"event":          v.Namespace(),
+								"plugin-name":    v.Name,
+								"plugin-version": v.Version,
+								"plugin-type":    core.PluginType(v.Type).String(),
+							}).Error("error validating dependencies")
```

Comment:
As above, consider using here `continue` instead of nesting the next things in _else_.